<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>treesifter Instructor’s Guide • treesiftr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="treesifter Instructor’s Guide">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">treesiftr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/00-Instuctor-Guide.html">treesifter Instructor's Guide</a>
    </li>
    <li>
      <a href="../articles/01-treesiftr.html">treesifter GUI</a>
    </li>
    <li>
      <a href="../articles/02-treesiftr-advanced.html">treesifter advanced</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>treesifter Instructor’s Guide</h1>
            
            <h4 class="date">2019-01-04</h4>
      
      
      <div class="hidden name"><code>00-Instuctor-Guide.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>treesiftr is an R package <span class="citation">(R Core Team 2018)</span> for visualizing the relationship between phylogenetic trees and phylogenetic data. Phylogenetic trees are crucial to the study of comparative biology, taxonomy, and evolution. However, understanding how to read a phylogenetic tree, and how a phylogenetic tree relates to underlying phylogenetic data, remains challenging.</p>
<p>treesiftr is intended to supplement classroom discussions of phylogeny. Sample slides that could be presented before the activity are included on the software’s GitHub <a href="https://github.com/wrightaprilm/treesiftr/tree/master/inst">repository</a>. I recommend using treesiftr after a preliminary discussion of tree thinking (such as that found in <span class="citation">Baum, Smith, and Donovan (2005)</span>), phylogeny estimation, and how to read phylogenetic trees. After the activity, come back together as a class to talk through the activity and the answers. For more novice classes, it may even be beneficial to do the first five questions, then regroup to discuss the answers before proceding.</p>
</div>
<div id="how-to-use-the-instructors-guide" class="section level1">
<h1 class="hasAnchor">
<a href="#how-to-use-the-instructors-guide" class="anchor"></a>How to Use the Instructor’s Guide</h1>
<p>treesiftr has two modes. The first is a graphical user interface (GUI). Exercise 01-treesiftr is meant to be used with the graphical interface. The second is the RStudio interface. 02-treesiftr-advanced is meant to be used in RStudio, with executeable code blocks.</p>
<p>The first part of the instructor guide concerns the use of the GUI. The second part explains the RStudio interface, with worked examples.</p>
<blockquote>
<p><em>Tip</em></p>
<p>Linked text goes to the glossary. If you see a term you don’t recognize, remember you can refresh your memory at the bottom of this worksheet!</p>
</blockquote>
<div id="treesiftr-instructors-guide" class="section level2">
<h2 class="hasAnchor">
<a href="#treesiftr-instructors-guide" class="anchor"></a>01-treesiftr Instructor’s Guide</h2>
<div id="learning-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning Objectives</h3>
<ul>
<li>Be able to count parsimony changes for the displayed characters on the estimated tree.
<ul>
<li>Questions: 1, 3</li>
</ul>
</li>
<li>Apply phylogenetic terminology, such as reversal, autapomorphy, and monophyletic, to discuss trait and lineage evolution.
<ul>
<li>Questions: 2, 4, 5</li>
</ul>
</li>
<li>Explain why two characters with the same parsimony score may not have the same likelihood score, and how the number of characters changes the likelihood and parsimony scores.
<ul>
<li>Questions: 6, 7, 8, 9</li>
</ul>
</li>
</ul>
<p>Navigate to the <a href="https://wrightaprilm.shinyapps.io/treesiftr_app/">treesiftr application</a>.</p>
<p>treesiftr works by subsetting a phylogenetic matrix using the <code>start</code> and <code>step</code> arguments. These are found on the left-hand side of the screen. The <code>start</code> argument controls where in the matrix you would like to begin visualizing characters. For example, a start value of 1 would indicate to begin visualizing characters from the first character in the matrix. The <code>step</code> value indicates how many characters at once to visualize. A step value of three would indicate characters should be viewed in threes. For example, if <code>start</code> = 1 and step = <code>3</code>, the visualization will show characters 1, 2, and 3. The start and end points of character visualizations are noted in the upper left-hand corner of the visualization. For example, in the below, visualization begins with the first character in the dataset. The first three characters (characters 1, 2, and 3) in the data matrix are visualized.</p>
<div class="figure">
<img src="img/IGwebinterface.png" alt="Web Interface example"><p class="caption">Web Interface example</p>
</div>
<p>There are also two checkboxes. The parsimony score checkbox uses Phangorn’s <span class="citation">(Schliep 2011, Schliep2017)</span> parsimony scoring algorithm to print the parsimony score of the visualized characters on the tree. The likelihood score is also calculated in Phangorn under the Mk model <span class="citation">(Lewis 2001)</span>. Leave these unchecked until otherwise noted.</p>
<p>treesiftr works by subsetting a phylogenetic matrix by user input. Then, a <a href="#glossary">parsimony tree</a> is constructed in Phangorn <span class="citation">(Schliep 2011, Schliep2017)</span> from the user-defined subset. The tree is <a href="#glossary">scored</a> under both parsimony and Lewis’ Mk model <span class="citation">(Lewis 2001)</span> for discrete character data. The data and tree are then visualized using ggtree <span class="citation">(Yu et al. 2017)</span>, based upon the ggplot2 package <span class="citation">(Wickham 2016)</span>. This application makes use of Shiny to provide a graphical interface, but there is a second included tutorial for more experienced users of the R statistical language.</p>
</div>
</div>
<div id="parsimony-questions" class="section level2">
<h2 class="hasAnchor">
<a href="#parsimony-questions" class="anchor"></a>Parsimony Questions</h2>
<ol style="list-style-type: decimal">
<li>Visualize characters 1 and 2. What is the parsimony score for this character set? Click “Do you want to print the <a href="#glossary">parsimony score?</a>” in the interface to check your answer.</li>
</ol>
<blockquote>
<p><em>Instructor’s Note:</em> The answer is two, as viewed in the below screenshot.</p>
</blockquote>
<div class="figure">
<img src="img/Q1.png" alt="Question One Answer"><p class="caption">Question One Answer</p>
</div>
<ol start="2" style="list-style-type: decimal">
<li>Visualize characters 2 and 3. What <a href="#glossary">monophyletic</a> group from tree 1 is no longer on this tree?</li>
</ol>
<blockquote>
<p><em>Instructor’s Note:</em> The answer is <em>Ursavus brevirhinus</em>, <em>Ballusia elmensis</em>, and <em>Ursavus primaevus</em>. <em>Ursavus primaevus</em> is no longer sister to <em>Ursavus<br>
brevirhinus</em> and <em>Ballusia elmensis</em> on the tree of character 2-3. Sample output follows.</p>
</blockquote>
<div class="figure">
<img src="img/Q2.png" alt="Question Two Answer"><p class="caption">Question Two Answer</p>
</div>
<ol start="3" style="list-style-type: decimal">
<li>What is the parsimony score of the 31-34 character set? Click “Do you want to print the parsimony score?” in the infterface to check your answer.</li>
</ol>
<blockquote>
<p><em>Instructor’s Note:</em> The answer is 7. The output you should see is below. <img src="img/Q3.png" alt="Question Three Answer"></p>
</blockquote>
<ol start="4" style="list-style-type: decimal">
<li>Which character, 8, 9 or 10, represents a <a href="#glossary">reversal</a>?</li>
</ol>
<blockquote>
<p><em>Instructor’s Note:</em> The root of the tree likely blue, with a change to brown on the branch leading to <em>Tremarctos</em>, <em>Melursus</em>, <em>Helarctos</em>, and <em>Ursus</em>. <em>Ursus thibetanus</em> posesses the ancestral blue state, and is the best candidate for reversal.</p>
</blockquote>
<div class="figure">
<img src="img/Q4.png" alt="Question Four Answer"><p class="caption">Question Four Answer</p>
</div>
<ol start="5" style="list-style-type: decimal">
<li>What information would we need to decide if the “1” state possesed by <em>Zaragocyon_daamsi</em> in character 52 is an <a href="#glossary">autapomorphy</a>?</li>
</ol>
<blockquote>
<p><em>Instructor’s Note:</em> To make this decision, we would want to know if there are other bears that are not sampled on this tree that posess this character.</p>
</blockquote>
<ol start="6" style="list-style-type: decimal">
<li>Click the switch that says “Do you want to print the likelihood score under the Mk model?” <em>and</em> the switch that says “Do you want to print the parsimony score?”. Do all characters with the same parsimony score have the same likelihood score?</li>
</ol>
<blockquote>
<p><em>Instructor’s Notes:</em> It is possible for characters with the same number of parsimony steps to have different likelihood scores.</p>
</blockquote>
<ol start="7" style="list-style-type: decimal">
<li>Compare characters 46-49 and 47-50. Why does set 47-50 have a better likelihood than 46-49?</li>
</ol>
<blockquote>
<p>Instructor’s Notes: Answers may vary. But generally, a character (character 46) that implies changes on a fairly short branch is replaced with one implying changes on longer branches.</p>
</blockquote>
<ol start="8" style="list-style-type: decimal">
<li>What is the relationship between the likelihood score and increasing the number of characters visualized?</li>
</ol>
<blockquote>
<p>Instructor’s Notes: The likelihood should become more negative (less likely)</p>
</blockquote>
<ol start="9" style="list-style-type: decimal">
<li>What is the minimum number adding a character can add to the parsimony score?</li>
</ol>
<blockquote>
<p>Instructor’s Notes: 1</p>
</blockquote>
<ol start="10" style="list-style-type: decimal">
<li>These trees are fully resolved. Based on your exploration of the data, does this degree of resolution make sense?</li>
</ol>
<blockquote>
<p>Instructor’s Notes: For some trees, we can see that there is <em>no</em> character data supporting the grouping. Parsimony software generally displays fully-resolved trees - this may be misleading if little to no data support the grouping!</p>
</blockquote>
</div>
</div>
<div id="treesiftr-advanced-instructors-guide" class="section level1">
<h1 class="hasAnchor">
<a href="#treesiftr-advanced-instructors-guide" class="anchor"></a>02-treesiftr-advanced Instructor’s Guide</h1>
<blockquote>
<p><em>Instructor’s Note:</em> A sample slide deck for more advanced audiences is included in the folder <code>inst/slides</code>. This slide deck is meant to be viewed in RStudio, as it has executeable code blocks. Hoever, it is also included in HTML format, if this is desireable for presentation. This is simply included as a sample of how one might choose to deliver a lecture to cover some of the phylogenetic terms in this activity. <em>It is not necessary to us this lecture to make use of this activity!</em></p>
</blockquote>
<p>treesiftr works by subsetting a phylogenetic matrix using the <code>start</code>, <code>stop</code> and <code>step</code> arguments. The <code>start</code> argument controls where in the matrix you would like to begin visualizing characters. For example, a start value of 1 would indicate to begin visualizing characters from the first character in the matrix. The <code>stop</code> value indicates what will be the first character in the last visualization. A stop value of 10 would indicate that the first character in the last visualization should be character 10. The <code>step</code> value indicates how many characters at once to visualize. A step value of three would indicate characters should be viewed in threes. For example, if <code>start</code> = 1, <code>stop</code> = 10, and step = <code>3</code>, 10 visualizations will be produced. The first will visualize characters 1, 2, and 3. The final will be characters 10, 11, and 12.</p>
<p>A <a href="#glossary">maximum parsimony</a> tree is then estimated from each dataset. The tree is <a href="#glossary">scored</a> under both parsimony and Lewis’ Mk model <span class="citation">(Lewis 2001)</span> for discrete character data. The data and tree are then visualized using ggtree, based upon the ggplot2 package. This application makes use of Shiny to provide a graphical interface, but in this tutorial, we will use the R Studio interface to visualize our data.</p>
<div id="learning-objectives-1" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-objectives-1" class="anchor"></a>Learning Objectives</h3>
<ul>
<li>Be able to count parsimony changes for the displayed characters on the estimated tree.
<ul>
<li>Questions: 1, 3</li>
</ul>
</li>
<li>Apply phylogenetic terminology, such as reversal, autapomorphy, and monophyletic, to discuss trait and lineage evolution.
<ul>
<li>Questions: 2, 4, 5</li>
</ul>
</li>
<li>Explain why two characters with the same parsimony score may not have the same likelihood score, and how the number of characters changes the likelihood and parsimony scores.
<ul>
<li>Questions: 6, 7, 8, 9</li>
</ul>
</li>
<li>Use indexing and slicing of data to view desired subsets of data.</li>
</ul>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Currently, treesiftr can be installed via the devtools <code>install_github</code> function. treesiftr has a number of required packages. Install and load the below.</p>
<blockquote>
<p><em>Instructor’s Note</em>: This does not auto-execute in the instructor’s guide.</p>
</blockquote>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co">#devtools::install_github("wrightaprilm/treesiftr")</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(treesiftr)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">library</span>(phangorn)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">library</span>(alignfigR)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">library</span>(ggtree)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">library</span>(ggplot2)</a></code></pre></div>
</div>
<div id="subsetting-the-phylogenetic-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#subsetting-the-phylogenetic-matrix" class="anchor"></a>Subsetting the phylogenetic matrix</h2>
<p>The first step to making a treesiftr visualization is to select the subset of the phylogenetic matrix that we would like to visualize. This is performed via a function called <code>generate_sliding</code>. The below command will subset the</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Locate package data and read alignment </span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">fdir &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"extdata"</span>, <span class="dt">package =</span> <span class="st">"treesiftr"</span>) </a>
<a class="sourceLine" id="cb2-3" data-line-number="3">aln_path &lt;-<span class="st"> </span><span class="kw">file.path</span>(fdir, <span class="st">"bears_fasta.fa"</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">bears &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(aln_path)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co"># Read in a starting phylogenetic tree</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">tree &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggtree/topics/reexports">read.tree</a></span>(<span class="kw">file.path</span>(fdir, <span class="st">"starting_tree.tre"</span>))</a>
<a class="sourceLine" id="cb2-7" data-line-number="7"></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co"># Generate our list of dataframe subsets</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">sample_df &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_sliding.html">generate_sliding</a></span>(bears, <span class="dt">start_char =</span> <span class="dv">1</span>, <span class="dt">stop_char =</span> <span class="dv">5</span>, <span class="dt">steps =</span> <span class="dv">1</span>)</a></code></pre></div>
<p>The result of this is a dataframe, shown below:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">sample_df</a></code></pre></div>
<pre><code>##   starting_val stop_val step_val
## 1            1        2        1
## 2            2        3        1
## 3            3        4        1
## 4            4        5        1
## 5            5        6        1</code></pre>
<p>This dataframe dispays the start character (the first character that will be visualized) and stop character (the final character that will be visualized).</p>
<p>We can then build trees from each subset:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears)</a></code></pre></div>
<pre><code>## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  1 nni operations 
## Final p-score 2 after  1 nni operations</code></pre>
<p>The above code saves the trees to a vector, but does not visualize them. They can be viauslized by naming the vector, like so:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">output_vector </a></code></pre></div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]
## NULL
## 
## [[3]]
## NULL
## 
## [[4]]
## NULL
## 
## [[5]]</code></pre>
<p><img src="00-Instuctor-Guide_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p><code>Phangorn</code> <span class="citation">(Schliep 2011, Schliep2017)</span> requires a starting tree to estimate a parsimony tree. We specify the tree we read in earlier for this purpose. The trees, which were generated with <code>ggtree</code> <span class="citation">(Yu et al. 2017)</span> have been saved to a vector, which can be displayed in its entirety, or subsetted to look at specific trees.</p>
</div>
<div id="questions" class="section level2">
<h2 class="hasAnchor">
<a href="#questions" class="anchor"></a>Questions</h2>
<ol style="list-style-type: decimal">
<li>Visualize characters 1 and 2. What is the parsimony score for this character set? Once you have answered this question,check your answer by adding the parismony score using the <code>pscore</code> argument:</li>
</ol>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears,</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">                                   <span class="dt">pscore =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  1 nni operations 
## Final p-score 2 after  1 nni operations</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">output_vector</a></code></pre></div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]
## NULL
## 
## [[3]]
## NULL
## 
## [[4]]
## NULL
## 
## [[5]]</code></pre>
<p><img src="00-Instuctor-Guide_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<blockquote>
<p><em>Instructor’s Note:</em> The answer is two, as viewed in the first visualization in the vector.</p>
</blockquote>
<ol start="2" style="list-style-type: decimal">
<li>Visualize characters 2 and 3. What <a href="#glossary">monophyletic</a> group from the tree of characters 1 and 2 is no longer on this tree?</li>
</ol>
<blockquote>
<p><em>Instructor’s Note:</em> The answer is <em>Ursavus brevirhinus</em>, <em>Ballusia elmensis</em>, and <em>Ursavus primaevus</em>. <em>Ursavus primaevus</em> is no longer sister to <em>Ursavus brevirhinus</em> and <em>Ballusia elmensis</em> on the tree of character 2-3.</p>
</blockquote>
<ol start="3" style="list-style-type: decimal">
<li>What is the parsimony score of the 31-34 character set?</li>
</ol>
<blockquote>
<p><em>Instructor’s Note:</em> The answer is 7. This can be checked like so:</p>
</blockquote>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">sample_df &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_sliding.html">generate_sliding</a></span>(bears, <span class="dt">start_char =</span> <span class="dv">31</span>, <span class="dt">stop_char =</span> <span class="dv">32</span>, <span class="dt">steps =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"></a>
<a class="sourceLine" id="cb13-3" data-line-number="3">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears,</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">                                   <span class="dt">pscore =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Final p-score 7 after  2 nni operations 
## Final p-score 9 after  2 nni operations</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">output_vector[<span class="dv">1</span>]</a></code></pre></div>
<pre><code>## [[1]]
## NULL</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Which character, 8, 9 or 10, represents a <a href="#glossary">reversal</a> from a <a href="#glossary">derived</a> state to <a href="#glossary">ancestral</a>?</li>
</ol>
<blockquote>
<p><em>Instructor’s Note:</em> The root of the tree likely blue, with a change to brown on the branch leading to <em>Tremarctos</em>, <em>Melursus</em>, <em>Helarctos</em>, and <em>Ursus</em>. <em>Ursus thibetanus</em> posesses the ancestral blue state, and is the best candidate for reversal. This can be verified like so:</p>
</blockquote>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">sample_df &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_sliding.html">generate_sliding</a></span>(bears, <span class="dt">start_char =</span> <span class="dv">8</span>, <span class="dt">stop_char =</span> <span class="dv">9</span>, <span class="dt">steps =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2"></a>
<a class="sourceLine" id="cb17-3" data-line-number="3">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears,</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">                                   <span class="dt">pscore =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Final p-score 4 after  0 nni operations 
## Final p-score 4 after  1 nni operations</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1">output_vector[<span class="dv">1</span>]</a></code></pre></div>
<pre><code>## [[1]]
## NULL</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>What information would we need to decide if the “1” state possesed by <em>Zaragocyon_daamsi</em> in character 52 is an <a href="#glossary">autapomorphy</a>?</li>
</ol>
<blockquote>
<p><em>Instructor’s Note:</em> To make this decision, we would want to know if there are other bears that are not sampled on this tree that posess this character. We can view the tree like so:</p>
</blockquote>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">sample_df &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_sliding.html">generate_sliding</a></span>(bears, <span class="dt">start_char =</span> <span class="dv">52</span>, <span class="dt">stop_char =</span> <span class="dv">53</span>, <span class="dt">steps =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"></a>
<a class="sourceLine" id="cb21-3" data-line-number="3">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears,</a>
<a class="sourceLine" id="cb21-4" data-line-number="4">                                   <span class="dt">pscore =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  0 nni operations</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">output_vector[<span class="dv">1</span>]</a></code></pre></div>
<pre><code>## [[1]]
## NULL</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>Do all characters with the same parsimony score have the same likelihood score? You can add the likelihood score to the visualization using the below code:</li>
</ol>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears, </a>
<a class="sourceLine" id="cb25-2" data-line-number="2">                                   <span class="dt">pscore =</span> <span class="ot">TRUE</span>, <span class="dt">lscore =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  0 nni operations</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">output_vector</a></code></pre></div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]</code></pre>
<p><img src="00-Instuctor-Guide_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<blockquote>
<p><em>Instructor’s Notes:</em> It is possible for characters with the same number of parsimony steps to have different likelihood scores.</p>
</blockquote>
<ol start="7" style="list-style-type: decimal">
<li>Compare characters 46-49 and 47-50. Why does set 47-50 have a better likelihood than 46-49?</li>
</ol>
<blockquote>
<p>Instructor’s Notes: Answers may vary. But generally, a character (character 46) that implies changes on a fairly short branch is replaced with one implying changes on longer branches.</p>
</blockquote>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1">sample_df &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_sliding.html">generate_sliding</a></span>(bears, <span class="dt">start_char =</span> <span class="dv">46</span>, <span class="dt">stop_char =</span> <span class="dv">49</span>, <span class="dt">steps =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb29-2" data-line-number="2"></a>
<a class="sourceLine" id="cb29-3" data-line-number="3">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears, </a>
<a class="sourceLine" id="cb29-4" data-line-number="4">                                   <span class="dt">pscore =</span> <span class="ot">TRUE</span>, <span class="dt">lscore =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## Final p-score 7 after  0 nni operations 
## Final p-score 7 after  0 nni operations 
## Final p-score 5 after  1 nni operations 
## Final p-score 5 after  1 nni operations</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">output_vector</a></code></pre></div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]
## NULL
## 
## [[3]]
## NULL
## 
## [[4]]</code></pre>
<p><img src="00-Instuctor-Guide_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<ol start="8" style="list-style-type: decimal">
<li><p>What is the relationship between the likelihood score and increasing the number of characters visualized? &gt; Instructor’s Notes: The likelihood should become more negative (less likely)</p></li>
<li><p>What is the minimum number adding a character can add to the parsimony score?</p></li>
</ol>
<blockquote>
<p>Instructor’s Notes: 1</p>
</blockquote>
<ol start="10" style="list-style-type: decimal">
<li>These trees are fully resolved. Based on your exploration of the data, does this degree of resolution make sense?</li>
</ol>
<blockquote>
<p>Instructor’s Notes: For some trees, we can see that there is <em>no</em> character data supporting the grouping. Parsimony software generally displays fully-resolved trees - this may be misleading if little to no data support the grouping!</p>
</blockquote>
</div>
<div id="glossary" class="section level2">
<h2 class="hasAnchor">
<a href="#glossary" class="anchor"></a>Glossary</h2>
<p><em>Ancestral State</em>: A character state possessed by the ancestor of a group</p>
<p><em>Autapomorphy</em>: A character state that is unique to a specific taxon.</p>
<p><em>Derived State</em>: A character state that is different from the ancestral state.</p>
<p><em>Likelihood Score</em>: The likelihood of the observed data under a specific model.</p>
<p><em>Maximum likelihood</em>: A phylogenetic optimatlity criterion under which phylogenetic data are modeled according to sets of assumptions. Under this criterion, the tree that has the best (“maximum”) likelihood score under the assumed model is to be preferred.</p>
<p><em>Maximum parsimony</em>: A phylogenetic optimality criterion. This criterion holds that the tree implying the fewest changes in the characters used to generate it should be preferred.</p>
<p><em>Monophyletic</em>: A group on a phylogeny of an ancestor and all of its descendents.</p>
<p><em>Parsimony Score</em>: The number of changes implied by a character on a tree.</p>
<p><em>Reversal</em>: A change from the derived state back to the ancestral state.</p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-Baum979">
<p>Baum, David A., Stacey DeWitt Smith, and Samuel S. S. Donovan. 2005. “The Tree-Thinking Challenge.” <em>Science</em> 310 (5750). American Association for the Advancement of Science: 979–80. <a href="https://doi.org/10.1126/science.1117727" class="uri">https://doi.org/10.1126/science.1117727</a>.</p>
</div>
<div id="ref-Lewis2001">
<p>Lewis, Paul O. 2001. “A Likelihood Approach to Estimating Phylogeny from Discrete Morphological Character Data.” <em>Systematic Biology</em> 50 (6). Oxford University Press: 913–25. <a href="https://doi.org/10.1080/106351501753462876" class="uri">https://doi.org/10.1080/106351501753462876</a>.</p>
</div>
<div id="ref-R">
<p>R Core Team. 2018. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-Schliep2011">
<p>Schliep, K.P. 2011. “Phangorn: Phylogenetic Analysis in R.” <em>Bioinformatics</em> 27 (4): 592–93. <a href="https://doi.org/10.1093/bioinformatics/btq706" class="uri">https://doi.org/10.1093/bioinformatics/btq706</a>.</p>
</div>
<div id="ref-ggplot2">
<p>Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York. <a href="http://ggplot2.org" class="uri">http://ggplot2.org</a>.</p>
</div>
<div id="ref-ggtree">
<p>Yu, Guangchuang, David Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. “Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.” <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628" class="uri">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#how-to-use-the-instructors-guide">How to Use the Instructor’s Guide</a><ul class="nav nav-pills nav-stacked">
<li><a href="#treesiftr-instructors-guide">01-treesiftr Instructor’s Guide</a></li>
      <li><a href="#parsimony-questions">Parsimony Questions</a></li>
      </ul>
</li>
      <li>
<a href="#treesiftr-advanced-instructors-guide">02-treesiftr-advanced Instructor’s Guide</a><ul class="nav nav-pills nav-stacked">
<li><a href="#installation">Installation</a></li>
      <li><a href="#subsetting-the-phylogenetic-matrix">Subsetting the phylogenetic matrix</a></li>
      <li><a href="#questions">Questions</a></li>
      <li><a href="#glossary">Glossary</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
