<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>treesifter GUI • treesiftr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="treesifter GUI">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">treesiftr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/00-Instuctor-Guide.html">treesifter Instructor's Guide</a>
    </li>
    <li>
      <a href="../articles/01-treesiftr.html">treesifter GUI</a>
    </li>
    <li>
      <a href="../articles/02-treesiftr-advanced.html">treesifter advanced</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>treesifter GUI</h1>
            
            <h4 class="date">2019-01-10</h4>
      
      
      <div class="hidden name"><code>01-treesiftr.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><a href="https://wrightaprilm.shinyapps.io/treesiftr_app/">treesiftr</a> is a Shiny application <span class="citation">(Chang et al. 2018)</span> for visualizing the relationship between phylogenetic trees and phylogenetic data. Phylogenetic trees are crucial to the study of comparative biology, taxonomy, and evolution. However, understanding how to read a phylogenetic tree, and how a phylogenetic tree relates to underlying phylogenetic data, remains challenging.</p>
<p>In today’s lab exercise, we will learn about a phylogenetic matrix, and then use the data in the matrix to visualize a phylogenetic tree.</p>
<blockquote>
<p><em>Tip</em></p>
<p>Linked text goes to the glossary. If you see a term you don’t recognize, remember you can refresh your memory at the bottom of this worksheet!</p>
</blockquote>
<div id="the-fossil-bear-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#the-fossil-bear-matrix" class="anchor"></a>The fossil bear matrix</h2>
<p>The data matrix included with treesiftr is a matrix of binary (“0” and “1”) characters compiled to estimate a topology of living and extinct bear species <span class="citation">(Abella et al. 2012)</span>. This matrix is fairly typical in size for a paleontological matrix, comprising 62 characters. It is, however, atypically complete, with only 18% missing data. In the following exercises, missing data will be represented by a thin black line. The “0” state will be represented in pale blue, and the “1” in brown.</p>
</div>
<div id="treesiftr" class="section level2">
<h2 class="hasAnchor">
<a href="#treesiftr" class="anchor"></a>treesiftr</h2>
<p>Navigate to the <a href="https://wrightaprilm.shinyapps.io/treesiftr_app/">treesiftr application</a>.</p>
<p>treesiftr works by subsetting a phylogenetic matrix using the <code>start</code> and <code>step</code> arguments. These are found on the left-hand side of the screen. The <code>start</code> argument controls where in the matrix you would like to begin visualizing characters. For example, a start value of 1 would indicate to begin visualizing characters from the first character in the matrix. The <code>step</code> value indicates how many characters at once to visualize. A step value of three would indicate characters should be viewed in threes. For example, if <code>start</code> = 1 and step = <code>3</code>, the visualization will show characters 1, 2, and 3. The start and end points of character visualizations are noted in the upper left-hand corner of the visualization.</p>
<p>There are also two checkboxes. Leave these unchecked until otherwise noted.</p>
<p>treesiftr works by subsetting a phylogenetic matrix by user input. Then, a <a href="#glossary">parsimony tree</a> is constructed in Phangorn <span class="citation">(Schliep 2011, Schliep2017)</span> from the user-defined subset. The tree is <a href="#glossary">scored</a> under both parsimony and Lewis’ Mk model <span class="citation">(Lewis 2001)</span> for discrete character data. The data and tree are then visualized using ggtree <span class="citation">(Yu et al. 2017)</span>, based upon the ggplot2 package <span class="citation">(Wickham 2016)</span>. This application makes use of Shiny to provide a graphical interface, but there is a second included tutorial for more experienced users of the R statistical language.</p>
</div>
<div id="questions" class="section level2">
<h2 class="hasAnchor">
<a href="#questions" class="anchor"></a>Questions</h2>
<ol style="list-style-type: decimal">
<li><p>Visualize characters 1 and 2. What is the parsimony score for this character set? Click “Do you want to print the <a href="#glossary">parsimony score?</a>” in the interface to check your answer. Uncheck the parsimony score option when you have completed this question.</p></li>
<li><p>Visualize characters 1 and 2. This time, click the “View random set of trees rather than an estimated tree” button in the interface. Are the parsimony scores larger or smaller than those of the character on an estimated tree? Does this make sense to you? Why? Remember to uncheck this box before continuing.</p></li>
<li><p>Visualize character set 2-3 and 3-4. What <a href="#glossary">monophyletic</a> group from the tree for characters 2-3 is no longer on the tree for characters 3-4?</p></li>
<li><p>View a tree of characters 8 - 10. Which character, 8, 9 or 10, represents a <a href="#glossary">reversal</a>?</p></li>
<li><p>What information would we need to decide if the “1” state possesed by <em>Zaragocyon_daamsi</em> in character 52 is an <a href="#glossary">autapomorphy</a>?</p></li>
<li><p>Click the switch that says “Do you want to print the likelihood score under the Mk model?” <em>and</em> the switch that says “Do you want to print the parsimony score?”. Does the character set 8, 9, and 10 have the same parsimony score as the character set 9, 10, and 11? How about the same likelihood score?</p></li>
<li><p>Compare characters 46-49 and 47-50. Why does set 47-50 have a better likelihood than 46-49?</p></li>
<li><p>What is the relationship between the likelihood score and increasing the number of characters visualized?</p></li>
<li><p>What is the minimum number adding a parsimony-informative character can add to the parsimony score?</p></li>
<li><p>These trees are fully resolved. Based on your exploration of the data, does this degree of resolution make sense? Take a look at characters 1-3, for example. Does displaying a fully resolved tree make sense for these characters?</p></li>
</ol>
</div>
<div id="glossary" class="section level2">
<h2 class="hasAnchor">
<a href="#glossary" class="anchor"></a>Glossary</h2>
<p><em>Ancestral State</em>: A character state possessed by the ancestor of a group</p>
<p><em>Autapomorphy</em>: A character state that is unique to a specific taxon.</p>
<p><em>Derived State</em>: A character state that is different from the ancestral state.</p>
<p><em>Likelihood Score</em>: The likelihood of the observed data under a specific model.</p>
<p><em>Maximum likelihood</em>: A phylogenetic optimatlity criterion under which phylogenetic data are modeled according to sets of assumptions. Under this criterion, the tree that has the best (“maximum”) likelihood score under the assumed model is to be preferred.</p>
<p><em>Maximum parsimony</em>: A phylogenetic optimality criterion. This criterion holds that the tree implying the fewest changes in the characters used to generate it should be preferred.</p>
<p><em>Monophyletic</em>: A group on a phylogeny of an ancestor and all of its descendents.</p>
<p><em>Parsimony Score</em>: The number of changes implied by a character on a tree.</p>
<p><em>Resolved</em>: A node that is bifucating, leaving two descendent lineages.</p>
<p><em>Reversal</em>: A change from the derived state back to the ancestral state.</p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-abella12">
<p>Abella, Juan, David M. Alba, Josep M. Robles, Alberto Valenciano, Cheyenn Rotgers, Raül Carmona, Plinio Montoya, and Jorge Morales. 2012. “<em>Kretzoiarctos</em> Gen. Nov., the Oldest Member of the Giant Panda Clade.” <em>PLoS One</em> 17: e48985.</p>
</div>
<div id="ref-shiny">
<p>Chang, Winston, Joe Cheng, JJ Allaire, Yihui Xie, and Jonathan McPherson. 2018. <em>Shiny: Web Application Framework for R</em>. <a href="https://CRAN.R-project.org/package=shiny" class="uri">https://CRAN.R-project.org/package=shiny</a>.</p>
</div>
<div id="ref-Lewis2001">
<p>Lewis, Paul O. 2001. “A Likelihood Approach to Estimating Phylogeny from Discrete Morphological Character Data.” <em>Systematic Biology</em> 50 (6). Oxford University Press: 913–25. <a href="https://doi.org/10.1080/106351501753462876" class="uri">https://doi.org/10.1080/106351501753462876</a>.</p>
</div>
<div id="ref-Schliep2011">
<p>Schliep, K.P. 2011. “Phangorn: Phylogenetic Analysis in R.” <em>Bioinformatics</em> 27 (4): 592–93. <a href="https://doi.org/10.1093/bioinformatics/btq706" class="uri">https://doi.org/10.1093/bioinformatics/btq706</a>.</p>
</div>
<div id="ref-ggplot2">
<p>Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York. <a href="http://ggplot2.org" class="uri">http://ggplot2.org</a>.</p>
</div>
<div id="ref-ggtree">
<p>Yu, Guangchuang, David Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. “Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.” <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628" class="uri">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#the-fossil-bear-matrix">The fossil bear matrix</a></li>
      <li><a href="#treesiftr">treesiftr</a></li>
      <li><a href="#questions">Questions</a></li>
      <li><a href="#glossary">Glossary</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
