<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>treesifter advanced • treesiftr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="treesifter advanced">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">treesiftr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/00-Instuctor-Guide.html">treesifter Instructor's Guide</a>
    </li>
    <li>
      <a href="../articles/01-treesiftr.html">treesifter GUI</a>
    </li>
    <li>
      <a href="../articles/02-treesiftr-advanced.html">treesifter advanced</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>treesifter advanced</h1>
            
            <h4 class="date">2019-01-02</h4>
      
      
      <div class="hidden name"><code>02-treesiftr-advanced.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>treesiftr is an R package <span class="citation">(R Core Team 2018)</span> for visualizing the relationship between phylogenetic trees and phylogenetic data. Phylogenetic trees are crucial to the study of comparative biology, taxonomy, and evolution. However, understanding how to read a phylogenetic tree, and how a phylogenetic tree relates to underlying phylogenetic data, remains challenging.</p>
<p>In today’s lab exercise, we will learn about the data in a phylogenetic matrix, and then use these data in the matrix to visualize a phylogenetic tree. We will use RStudio <span class="citation">(RStudio Team 2015)</span> to conduct our exercise.</p>
<blockquote>
<p><em>Tip</em></p>
<p>Linked text goes to the glossary. If you see a term you don’t recognize, remember you can refresh your memory at the bottom of this worksheet!</p>
</blockquote>
<div id="the-fossil-bear-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#the-fossil-bear-matrix" class="anchor"></a>The fossil bear matrix</h2>
<p>The data matrix included with treesiftr is a matrix of binary (“0” and “1”) characters compiled to estimate a topology of living and extinct bear species <span class="citation">(Abella et al. 2012)</span>. This matrix is fairly typical in size for a paleontological matrix, comprising 62 characters. It is, however, atypically complete, with only 18% missing data. In the following exercises, missing data will be represented by a thin black line. The “0” state will be represented in pale blue, and the “1” in brown.</p>
</div>
<div id="treesiftr" class="section level2">
<h2 class="hasAnchor">
<a href="#treesiftr" class="anchor"></a>treesiftr</h2>
<p>treesiftr works by subsetting a phylogenetic matrix using the <code>start</code>, <code>stop</code> and <code>step</code> arguments. The <code>start</code> argument controls where in the matrix you would like to begin visualizing characters. For example, a start value of 1 would indicate to begin visualizing characters from the first character in the matrix. The <code>stop</code> value indicates what will be the first character in the last visualization. A stop value of 10 would indicate that the first character in the last visualization should be character 10. The <code>step</code> value indicates how many characters at once to visualize. A step value of three would indicate characters should be viewed in threes. For example, if <code>start</code> = 1, <code>stop</code> = 10, and step = <code>3</code>, 10 visualizations will be produced. The first will visualize characters 1, 2, and 3. The final will be characters 10, 11, and 12.</p>
<p>A <a href="#glossary">maximum parsimony</a> tree is then estimated from each dataset. The tree is <a href="#glossary">scored</a> under both parsimony and Lewis’ Mk model <span class="citation">(Lewis 2001)</span> for discrete character data. The data and tree are then visualized using ggtree, based upon the ggplot2 package. This application makes use of Shiny to provide a graphical interface, but in this tutorial, we will use the R Studio interface to visualize our data.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>Currently, treesiftr can be installed via the devtools <code>install_github</code> function. treesiftr has a number of required packages. Install and load the below.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">install_github</a></span>(<span class="st">"wrightaprilm/treesiftr"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">library</span>(treesiftr)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw">library</span>(phangorn)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">library</span>(alignfigR)</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="kw">library</span>(ggtree)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="kw">library</span>(ggplot2)</a></code></pre></div>
</div>
<div id="subsetting-the-phylogenetic-matrix" class="section level2">
<h2 class="hasAnchor">
<a href="#subsetting-the-phylogenetic-matrix" class="anchor"></a>Subsetting the phylogenetic matrix</h2>
<p>The first step to making a treesiftr visualization is to select the subset of the phylogenetic matrix that we would like to visualize. This is performed via a function called <code>generate_sliding</code>. The below command will subset the</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Locate package data and read alignment </span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">fdir &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="dt">package =</span> <span class="st">"treesiftr"</span>) </a>
<a class="sourceLine" id="cb2-3" data-line-number="3">aln_path &lt;-<span class="st"> </span><span class="kw">file.path</span>(fdir, <span class="st">"bears_fasta.fa"</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">bears &lt;-<span class="st"> </span>alignfigR<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/alignfigR/topics/read_alignment">read_alignment</a></span>(aln_path)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co"># Read in a starting phylogenetic tree</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">tree &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/ape/topics/read.tree">read.tree</a></span>(<span class="kw">file.path</span>(fdir, <span class="st">"starting_tree.tre"</span>))</a>
<a class="sourceLine" id="cb2-7" data-line-number="7"></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co"># Generate our list of dataframe subsets</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9">sample_df &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_sliding.html">generate_sliding</a></span>(bears, <span class="dt">start_char =</span> <span class="dv">1</span>, <span class="dt">stop_char =</span> <span class="dv">5</span>, <span class="dt">steps =</span> <span class="dv">1</span>)</a></code></pre></div>
<p>The result of this is a dataframe, shown below:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">sample_df</a></code></pre></div>
<pre><code>##   starting_val stop_val step_val
## 1            1        2        1
## 2            2        3        1
## 3            3        4        1
## 4            4        5        1
## 5            5        6        1</code></pre>
<p>This dataframe dispays the start character (the first character that will be visualized) and stop character (the final character that will be visualized).</p>
<p>We can then build trees from each subset:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears)</a></code></pre></div>
<pre><code>## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  0 nni operations 
## Final p-score 2 after  1 nni operations 
## Final p-score 2 after  1 nni operations</code></pre>
<p>The above code saves the trees to a vector, but does not visualize them. They can be visualized by naming the vector, like so:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">output_vector </a></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="02-treesiftr-advanced_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<pre><code>## 
## [[2]]</code></pre>
<p><img src="02-treesiftr-advanced_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
<pre><code>## 
## [[3]]</code></pre>
<p><img src="02-treesiftr-advanced_files/figure-html/unnamed-chunk-5-3.png" width="700"></p>
<pre><code>## 
## [[4]]</code></pre>
<p><img src="02-treesiftr-advanced_files/figure-html/unnamed-chunk-5-4.png" width="700"></p>
<pre><code>## 
## [[5]]</code></pre>
<p><img src="02-treesiftr-advanced_files/figure-html/unnamed-chunk-5-5.png" width="700"></p>
<p><code>Phangorn</code> <span class="citation">(Schliep 2011, Schliep2017)</span> requires a starting tree to estimate a parsimony tree. We specify the tree we read in earlier for this purpose. The trees, which were generated with <code>ggtree</code> <span class="citation">(Yu et al. 2017)</span> have been saved to a vector, which can be displayed in its entirety, or subsetted to look at specific trees.</p>
</div>
<div id="questions" class="section level2">
<h2 class="hasAnchor">
<a href="#questions" class="anchor"></a>Questions</h2>
<ol style="list-style-type: decimal">
<li>Visualize characters 1 and 2. What is the parsimony score for this character set? Once you have answered this question,check your answer by adding the parismony score using the <code>pscore</code> argument:</li>
</ol>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears,</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">                                   <span class="dt">pscore =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li><p>Visualize characters 2 and 3. What <a href="#glossary">monophyletic</a> group from the tree of characters 1 and 2 is no longer on this tree?</p></li>
<li><p>What is the parsimony score of the 31-34 character set?</p></li>
<li><p>Which character, 8, 9 or 10, represents a <a href="#glossary">reversal</a> from a <a href="#glossary">derived</a> state to <a href="#glossary">ancestral</a>?</p></li>
<li><p>What information would we need to decide if the “1” state possesed by <em>Zaragocyon_daamsi</em> in character 52 is an <a href="#glossary">autapomorphy</a>?</p></li>
<li><p>Do all characters with the same parsimony score have the same likelihood score? You can add the likelihood score to the visualization using the below code:</p></li>
</ol>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">output_vector &lt;-<span class="st"> </span><span class="kw"><a href="../reference/generate_tree_vis.html">generate_tree_vis</a></span>(<span class="dt">sample_df =</span> sample_df, <span class="dt">alignment =</span>                                                     aln_path, <span class="dt">tree =</span> tree, <span class="dt">phy_mat =</span> bears, </a>
<a class="sourceLine" id="cb14-2" data-line-number="2">                                   <span class="dt">pscore =</span> <span class="ot">TRUE</span>, <span class="dt">lscore =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<ol start="7" style="list-style-type: decimal">
<li><p>Compare characters 46-49 and 47-50. Why does set 47-50 have a better likelihood than 46-49?</p></li>
<li><p>What is the relationship between the likelihood score and increasing the number of characters visualized?</p></li>
<li><p>What is the minimum number adding a character can add to the parsimony score?</p></li>
<li><p>These trees are fully resolved. Based on your exploration of the data, does this degree of resolution make sense?</p></li>
</ol>
</div>
<div id="glossary" class="section level2">
<h2 class="hasAnchor">
<a href="#glossary" class="anchor"></a>Glossary</h2>
<p><em>Ancestral State</em>: A character state possessed by the ancestor of a group</p>
<p><em>Autapomorphy</em>: A character state that is unique to a specific taxon.</p>
<p><em>Derived State</em>: A character state that is different from the ancestral state.</p>
<p><em>Likelihood Score</em>: The likelihood of the observed data under a specific model.</p>
<p><em>Maximum likelihood</em>: A phylogenetic optimatlity criterion under which phylogenetic data are modeled according to sets of assumptions. Under this criterion, the tree that has the best (“maximum”) likelihood score under the assumed model is to be preferred.</p>
<p><em>Maximum parsimony</em>: A phylogenetic optimality criterion. This criterion holds that the tree implying the fewest changes in the characters used to generate it should be preferred.</p>
<p><em>Monophyletic</em>: A group on a phylogeny of an ancestor and all of its descendents.</p>
<p><em>Parsimony Score</em>: The number of changes implied by a character on a tree.</p>
<p><em>Reversal</em>: A change from the derived state back to the ancestral state.</p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-abella12">
<p>Abella, Juan, David M. Alba, Josep M. Robles, Alberto Valenciano, Cheyenn Rotgers, Raül Carmona, Plinio Montoya, and Jorge Morales. 2012. “<em>Kretzoiarctos</em> Gen. Nov., the Oldest Member of the Giant Panda Clade.” <em>PLoS One</em> 17: e48985.</p>
</div>
<div id="ref-Lewis2001">
<p>Lewis, Paul O. 2001. “A Likelihood Approach to Estimating Phylogeny from Discrete Morphological Character Data.” <em>Systematic Biology</em> 50 (6). Oxford University Press: 913–25.</p>
</div>
<div id="ref-R">
<p>R Core Team. 2018. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-RStudio">
<p>RStudio Team. 2015. <em>RStudio: Integrated Development Environment for R</em>. Boston, MA: RStudio, Inc. <a href="http://www.rstudio.com/" class="uri">http://www.rstudio.com/</a>.</p>
</div>
<div id="ref-Schliep2011">
<p>Schliep, K.P. 2011. “Phangorn: Phylogenetic Analysis in R.” <em>Bioinformatics</em> 27 (4): 592–93. <a href="https://doi.org/10.1093/bioinformatics/btq706" class="uri">https://doi.org/10.1093/bioinformatics/btq706</a>.</p>
</div>
<div id="ref-ggtree">
<p>Yu, Guangchuang, David Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. “Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.” <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628" class="uri">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#the-fossil-bear-matrix">The fossil bear matrix</a></li>
      <li><a href="#treesiftr">treesiftr</a></li>
      <li><a href="#installation">Installation</a></li>
      <li><a href="#subsetting-the-phylogenetic-matrix">Subsetting the phylogenetic matrix</a></li>
      <li><a href="#questions">Questions</a></li>
      <li><a href="#glossary">Glossary</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
